impl/src/main/java/org/jboss/weld/injection/producer/NoopLifecycleCallbackInvoker.java

```diff
@@ -55,6 +55,12 @@ public boolean hasPreDestroyMethods() {
 
     @Override
     public boolean hasPostConstructMethods() {
-        return true; // we cannot say no for sure
+        return hasPostConstructCallback();
```

Comment:
I still don't comprehend this approach - the class is used for interceptors, I get that much. We don't want/need post contruct callbacks there.
What I don't get, is why we still invoke them? Putting `System.out` in `postConstruct` will trigger it. That is IMO not intended